test/extensions/filters/http/geoip/geoip_filter_test.cc

```diff
@@ -0,0 +1,345 @@
+#include "envoy/extensions/filters/http/geoip/v3/geoip.pb.h"
+
+#include "source/common/network/address_impl.h"
+#include "source/common/network/utility.h"
+#include "source/extensions/filters/http/geoip/geoip_filter.h"
+
+#include "test/extensions/filters/http/geoip/mocks.h"
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+#include "test/mocks/stats/mocks.h"
+#include "test/test_common/registry.h"
+#include "test/test_common/test_runtime.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::DoAll;
+using testing::ReturnRef;
+using testing::SaveArg;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Geoip {
+namespace {
+
+MATCHER_P2(HasExpectedHeader, expected_header, expected_value, "") {
+  auto request_headers = static_cast<Http::TestRequestHeaderMapImpl>(arg);
+  EXPECT_TRUE(request_headers.has(expected_header));
```

Comment:
In general using EXPECT inside a matcher is likely to result in some confusing behaviors on test failure. Better to do `if (fail condition) {*result_listener<<explanation; return false;}`

In this case because it's using get, the expected_header param can't be a matcher, but still worth using `testing::ExplainMatchResult` for the comparison of expected_value so that sub-matchers are an option.

Reply:
You changed it to not need the `EXPECT_TRUE`, but forgot to delete the now unnecessary `EXPECT_TRUE`. :)